tests/testthat/test-PackageFunctionReporter.R

```diff
@@ -0,0 +1,167 @@
+context("Package Function Reporter Tests")
+
+##### TEST SET UP #####
+
+rm(list = ls())
+# Configure logger (suppress all logs in testing)
+# expect_silents only work with this logger turned off; only alerts with warnings
+loggerOptions <- futile.logger::logger.options()
+if (!identical(loggerOptions, list())){
+  origLogThreshold <- loggerOptions[[1]][['threshold']]
+} else {
+  origLogThreshold <- futile.logger::INFO
+}
+futile.logger::flog.threshold(0)
+
+##### TESTS #####
+
+## Structure Available ##
+
+test_that('PackageFunctionReporter structure is as expected', {
+  
+  expect_named(object = PackageFunctionReporter$public_methods
+               , expected = c(
+                 "calculate_metrics", 
+                 "extract_network", 
+                 "clone"
+               )
+               , info = "Available public methods for PackageFunctionReporter not as expected."
+  )
+  
+  expect_named(object = PackageFunctionReporter$public_fields
+               , expected = NULL
+               , info = "Available public fields for PackageFunctionReporter not as expected."
+  )
+  
+  expect_named(object = PackageFunctionReporter$private_methods
+               , expected = c("package_test_coverage")
+               , info = "Available private methods for PackageFunctionReporter not as expected."
+  )
+  
+  expect_named(object = PackageFunctionReporter$private_fields
+               , expected = NULL
+               , info = "Available private fields for PackageFunctionReporter not as expected."
+  )
+  
+})
+
+### USAGE OF PUBLIC AND PRIVATE METHODS AND FIELDS
+
+test_that('PackageFunctionReporter Methods Work', {
+  testObj <- PackageFunctionReporter$new()
+  
+  # inherited set_package
+  expect_silent(object = testObj$set_package(packageName = "baseballstats"
+                                             # Covr only works on source code. find.package path will not work
+                                             , packagePath = system.file("baseballstats",package="pkgnet")
```

Comment:
hey I have an idea....

given that this worked, couldn't we test _any_ package given just the package name? Like couldn't we always take that `packagePath` to be `system.file(package=thePackageYouAreLookingAt)`?

If so, that would mean that we could implement test coverage stuff without needing to add `packagePath` to the interface. That would be pretty baller

Reply:
That should work.  However, that also means adding a significant amount of run time by default: sub 1 second vs 20+ seconds for some well covered packages.  Thoughts?